src/core/src/core_logic/RebootManager.py

```diff
@@ -104,6 +104,11 @@ def start_reboot_if_required_and_time_available(self, current_time_available):
             self.composite_logger.log(" - There was no reboot pending detected. Reboot is being skipped as it's not required, as per patch installation configuration (" + str(Constants.REBOOT_IF_REQUIRED) + ").")
             return False
 
+        # prevent repeated reboots
+        if self.reboot_setting == Constants.REBOOT_ALWAYS and not reboot_pending and self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
+            self.composite_logger.log(" - At least one reboot has occurred, and there's no reboot pending, so the conditions for the 'Reboot Always' setting is fulfilled and reboot won't be repeated.")
+            return False
```

Comment:
Should we also update rebootPending value in this case?

Reply:
reboot_pending is from the source of truth on the VM.. we don't manipulate it.